modules/Router.js

```diff
@@ -54,6 +59,27 @@ const Router = React.createClass({
   },
 
   componentWillMount() {
+    const { history } = this.props
+    if (isDeprecatedHistory(history)) {
+      this.setupDeprecatedHistory()
+    } else {
+      this.setupTransitionManager()
+    }
+  },
+
+  setupTransitionManager() {
+    const { history, routes, children } = this.props
+    this.transitionManager = createTransitionManager(history, createRoutes(routes || children))
+    this.transitionManager.listen((error, state) => {
+      if (error) {
+        this.handleError(error)
+      } else {
+        this.setState(state, this.props.onUpdate)
+      }
+    })
+  },
```

Comment:
I agree, but for backwards compat we need `history` on context that looks just like it used to. So in v+2 we can do that.

Backwards compat, amirite?!